packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -1267,6 +1267,50 @@ void main() {
       }
     });
 
+    controller.value = controller.value.copyWith(isPlaying: true);
+    fakeVideoEventStream.add(VideoEvent(eventType: VideoEventType.completed));
+  });
+
+  test(
+      'isCompleted seeks position to max duration and pauses the video first if is playing',
```

Comment:
Isn't the thing you want to fix that you want to *not* call pause if it's *not* playing? That path would also need a test.